htdocs/accountancy/customer/lines.php

```diff
@@ -191,6 +191,7 @@
 $sql .= " INNER JOIN ".MAIN_DB_PREFIX."societe as s ON s.rowid = f.fk_soc";
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_country as co ON co.rowid = s.fk_pays ";
 $sql .= " WHERE fd.fk_code_ventilation > 0";
+$sql .= " AND l.total_ht <> 0";
```

Comment:
Some record may remains with status "to dispatch in accountancy" all the time.
Why not adding the filter during export of FEC file ?

Reply:
A l'utilisation chez un client qui a environ 6000 lignes a zéro pour des frais de port tous les mois. Aucun problème constaté, il n'y a pas de protection dans Dolibarr pour dire si l'opération est complétement répartie ou non pour son transfert en comptabilité. 
Après, on peut faire le tri directement au niveau de l'export, nous sommes d'accord mais quel est l'intérêt d'avoir en comptabilité, des lignes, voir des milliers de lignes à zéro...
Si l'administration Française s'en fiche, je les comprends et je les suis.

Reply:
Maybe an option ?

Reply:
Curious.
During transfer into accountancy, there is a test 
`if ($errorforinvoice[$key] == 'somelinesarenotbound')`
To be sure to not try to transfer if binding is not complete.

A checked and a test is done on
`AND fd.total_ttc <> 0`
to exclude line when making this check. This is the reason why you don't experience trouble, because on your example total_ht is 0 but total_ttc is probably also 0 so excluded from the check.

For homogeneous rules, we should test on same rule
total_ttc <> 0 or total_ht <> 0
Sometimes we make invoice line with just VAT so total_ht = 0 and vat is not null and total_ttc <> 0. We should keep this line. What do you think about inverting your test
total_ht <> 0 into total_ttc <> 0 ?

Reply:
@aspangaro What do you think about inverting your test
total_ht <> 0 into total_ttc <> 0 ? (see previous comment)